src/pint/modelutils.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from pint.models.astrometry import AstrometryEquatorial, AstrometryEcliptic
+
+
+def convert_to_equatorial(model, force=False):
+    r"""Converts Astrometry model component, Ecliptic to Equatorial
+
+    Parameters
+    ----------
+    model: `pint.models.TimingModel` object
+        current model with AstrometryEcliptic component
+    force: boolean, optional
+        will force conversion even if an equatorial component is already present
+
+    Returns
+    -------
+    model
+        new model with AstrometryEquatorial component
+    """
+
+    if not ("AstrometryEquatorial" in model.components) or force:
+        if "AstrometryEquatorial" in model.components:
+            logger.warning(
+                "Equatorial coordinates already present but re-calculating anyway"
+            )
+
+        if "AstrometryEcliptic" in model.components:
+
+            c = model.coords_as_ICRS()
+            a = AstrometryEquatorial()
+
+            a.POSEPOCH = model.POSEPOCH
+            a.PX = model.PX
+
+            a.RAJ.quantity = c.ra
+            a.DECJ.quantity = c.dec
+            a.PMRA.quantity = c.pm_ra_cosdec
+            a.PMDEC.quantity = c.pm_dec
+
+            model.add_component(a)
+            model.remove_component("AstrometryEcliptic")
+
+            model.setup()
+            model.validate()
+
+        else:
+            raise AttributeError(
+                "Requested conversion to equatorial coordinates, but no alternate coordinates found"
+            )
+
+    else:
+        logger.debug("Equatorial coordinates already present; not re-calculating")
+
+    return model
+
+
+def convert_to_ecliptic(model, force=False):
```

Comment:
Same but this is from equatorial to pulsar ecliptic.

Reply:
Fixed (see above).